hashing/src/lib.rs

```diff
@@ -89,32 +128,32 @@ impl Digest {
         self.0.to_vec()
     }
 
-    /// Hashes a `Vec` of `Digest`s into a single `Digest` by constructing a [Merkle tree][1].
-    /// Reduces pairs of elements in the `Vec` by repeatedly calling [`Digest::hash_pair`]. This
-    /// hash procedure is suited to hashing `BTree`s.
+    /// Hashes a `impl IntoIterator` of [`Digest`]s into a single [`Digest`] by
+    /// constructing a [Merkle tree][1]. Reduces pairs of elements in the collection by repeatedly
+    /// calling [Digest::hash_pair].
     ///
-    /// The pattern of hashing is as follows.  It is akin to [graph reduction][2]:
+    /// The pattern of hashing is akin to [graph reduction][2] and is described
+    /// in the documentation of [`.tree_fold1()`](Itertools::tree_fold1).
     ///
-    /// ```text
-    /// a b c d e f
-    /// |/  |/  |/
-    /// g   h   i
-    /// | /   /
-    /// |/   /
-    /// j   k
-    /// | /
-    /// |/
-    /// l
-    /// ```
```

Comment:
@rafal-ch Don't get rid of this documentation.

Reply:
Documentation re-added.

Reply:
For the record, the thing that was there before is pretty much identical, but this has nicer ascii art...